site/src/components/Tabs/Tabs.tsx

```diff
@@ -1,78 +1,94 @@
+
+export const Tabs: FC<TabsProps> = ({ active, ...htmlProps }) => {
+  const theme = useTheme();
 
-export const Tabs: FC<PropsWithChildren> = ({ children }) => {
   return (
-    <div
-      css={(theme) => ({
-        borderBottom: `1px solid ${theme.palette.divider}`,
-        marginBottom: 40,
-      })}
-    >
-      <Margins
+    <TabsContext.Provider value={{ active }}>
+      <div
         css={{
-          display: "flex",
-          alignItems: "center",
-          gap: 2,
+          borderBottom: `1px solid ${theme.palette.divider}`,
         }}
-      >
-        {children}
-      </Margins>
-    </div>
+        {...htmlProps}
+      />
+    </TabsContext.Provider>
+  );
+};
+
+type TabsListProps = HTMLAttributes<HTMLDivElement>;
+
+export const TabsList: FC<TabsListProps> = (props) => {
+  return (
+    <div
+      role="tablist"
+      css={{
+        display: "flex",
+        alignItems: "baseline",
+      }}
+      {...props}
+    />
   );
 };
 
-interface TabLinkProps extends NavLinkProps {
-  className?: string;
-}
+type TabLinkProps = LinkProps & {
+  value: string;
+};
 
-export const TabLink: FC<TabLinkProps> = ({
-  className,
-  children,
-  ...linkProps
-}) => {
-  const tabLink = useClassName(classNames.tabLink, []);
-  const activeTabLink = useClassName(classNames.activeTabLink, []);
+export const TabLink: FC<TabLinkProps> = ({ value, ...linkProps }) => {
+  const tabsContext = useContext(TabsContext);
+
+  if (!tabsContext) {
+    throw new Error("Tab only can be used inside of Tabs");
+  }
+
+  const isActive = tabsContext.active === value;
 
   return (
-    <NavLink
-      className={({ isActive }) =>
-        cx([tabLink, isActive && activeTabLink, className])
-      }
+    <Link
       {...linkProps}
-    >
-      {children}
-    </NavLink>
+      css={[styles.tabLink, isActive ? styles.activeTabLink : ""]}
+    />
   );
 };
 
-const classNames = {
-  tabLink: (css, theme) => css`
-    text-decoration: none;
-    color: ${theme.palette.text.secondary};
-    font-size: 14px;
-    display: block;
-    padding: 0 16px 16px;
-
-    &:hover {
-      color: ${theme.palette.text.primary};
-    }
-  `,
-  activeTabLink: (css, theme) => css`
-    color: ${theme.palette.text.primary};
-    position: relative;
-
-    &:before {
-      content: "";
-      left: 0;
-      bottom: 0;
-      height: 2px;
-      width: 100%;
-      background: ${theme.palette.primary.main};
-      position: absolute;
-    }
-  `,
-} satisfies Record<string, ClassName>;
+const styles = {
+  tabLink: (theme) => ({
+    textDecoration: "none",
+    color: theme.palette.text.secondary,
+    fontSize: 14,
+    display: "block",
+    padding: `${TAB_PADDING_Y}px 16px`,
+    fontWeight: 500,
+    lineHeight: "1",
+
+    "&:hover": {
+      color: theme.palette.text.primary,
+    },
+  }),
+  activeTabLink: (theme) => ({
+    color: theme.palette.text.primary,
+    position: "relative",
+
+    "&:before": {
```

Comment:
Yes. In the way you suggested it does not vertically align the content correctly since it is off by 1px at the bottom so you also would need to add a border to the top which starts to make the CSS more "complicated" IMO. I just think the before is easier to understand and implement.